modules/jfx.incubator.richtext/src/main/java/jfx/incubator/scene/control/richtext/model/RichParagraph.java

```diff
@@ -112,8 +116,21 @@ public Supplier<Region> getParagraphRegion() {
      */
     public abstract String getPlainText();
 
-    // this method could be made public, as long as the returned list is made immutable
-    abstract List<StyledSegment> getSegments();
+    /**
+     * Returns the number of segments in the paragraph.
+     * @return the number of segments
+     * @since 26
+     */
+    public abstract int getSegmentCount();
+
+    /**
+     * Returns the segment at the specified index.
+     * @param index the segment index
+     * @return the segment
+     * @throws IndexOutOfBoundsException if the index is outside of the range {@code 0 ... getSegmentCount()-1} (inclusive)
+     * @since 26
+     */
+    public abstract StyledSegment getSegment(int index);
```

Comment:
A `List<StyledSegment> getSegments()` method would be more convenient to use. You could either return an unmodifiable view of the list or make a shallow copy.

What is the reason for separate `getSegmentCount` and `getSegment(index)` methods?

Reply:
In theory, `List<StyledSegment> getSegments()` can be added as a convenience method which returns an unmodifiable copy.